sycamore/transforms/extract_table.py

```diff
@@ -108,3 +114,152 @@ def extract_tables(self, document: Document) -> Document:
         tables = self._extract(document)
         document.elements = document.elements + tables
         return document
+
+
+class CachedTextractTableExtractor(TableExtractor):
+    """
+    TextractTableExtractor with S3 based cache support
+
+    For each document, CachedTextractTableExtractor follows a series of steps to do table extraction
+     1. if table pages not exist, leave document unchanged
+     2. if there is a cache hit for current doc, if true, return the cached extracted tables from S3
+     3. if a cache miss happens, clip pages which contain tables and run table extraction using textractor,
+      also update cache accordingly.
+
+    Cached table is in Json format as below:
+    {
```

Comment:
So everything is in a single file? Do you have a sense of how far this will scale? For datasets with thousands of big tables, wouldn't downloading/updating the cache eventually become a bottleneck. I don't want to boil the ocean now, but curious what your thoughts are.

Reply:
not sure, how big it scales on disk, but seems today textractor just put all the result in memory, I don't have a lot of concern on the file size for now.